src/XrdCl/XrdClZipArchive.cc

```diff
@@ -430,6 +430,19 @@ namespace XrdCl
                                         else
                                           std::tie( cdvec, cdmap ) = CDFH::Parse( buff, eocd->cdSize, eocd->nbCdRec );
                                         log->Dump( ZipMsg, "[0x%x] CD records parsed.", this );
+                                        uint64_t sumSize = 0;
```

Comment:
My understanding is you are summing the "compressed" and "uncompressed" size and making sure this does not exceed the archive size, in order to make sure we are not reading past the end of the buffer. Now the buffer will contain either compressed data (in case the archive was compressed) or uncompressed data (in case it was not compressed), so it doesn't make sense to check both. If it is compressed check only the sum of "compressed" file sizes, and if it is uncompressed the sum of "uncompressed" file sizes. In practice (AFAIR), if the file is NOT compressed the value of "compressed" size and "uncompressed" size will be the same (please double check in the documentation: https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT), so in the end you can get away with always only checking "compressed" size.